### PR TITLE
PURCHASE-1454 Open web view target='_blank' links by pushing to navigation stack

### PR DESCRIPTION
**Pull request by ds300**
Fixes [PURCHASE-1454](https://artsyproduct.atlassian.net/browse/PURCHASE-1454)

For some reason this navigation messes with the close button's position persistently. We already have a QA ticket for that bug on notion because it's affecting the auction views too, so I'm happy to tackle it separately. What do you think @alloy?

### CODE REVIEW

**Comment by alloy** [comment]
Can you please link to the actual jira ticket?

> For some reason this navigation messes with the close button's position persistently. We already have a QA ticket for that bug on notion because it's affecting the auction views too, so I'm happy to tackle it separately. What do you think @alloy?

I concur 👍

**Comment by alloy** [comment]
Can you add some test coverage? But otherwise this is great 👌

**Comment by ds300** [comment]
@alloy tests added 🍏

File: Artsy_Tests/View_Controller_Tests/Web_Browsing/ARExternalWebBrowserViewControllerTests.m [hunk 1]
```diff
@@ -75,5 +76,33 @@ @interface ARExternalWebBrowserViewController (Tests) <UIScrollViewDelegate>
...
+
+  [[[mockRequest stub] andReturn:[NSURL URLWithString:@"https://artsy.net/conditions-of-sale"]] URL];
+
+  [[[mockAction stub] andReturn:mockFrame] targetFrame];
+  [[[mockAction stub] andReturn:mockRequest] request];
+
+  UINavigationController *nav = [[UINavigationController alloc] initWithRootViewController:vc];
+
+  id navMock = [OCMockObject partialMockForObject:nav];
+  [[navMock expect] pushViewController:OCMOCK_ANY animated:OCMOCK_ANY];
```
Comment (alloy) [inline Artsy_Tests/View_Controller_Tests/Web_Browsing/ARExternalWebBrowserViewControllerTests.m]: For next time, there’s also a OCMock matcher that takes a block and allows you to assert against the given argument.

**Comment by alloy** [comment]
Great stuff, thanks @ds300 👍 

#MergeOnGreen

**Pull request merged**